src/lib/helpers/trim.ts

```diff
@@ -1,6 +1,30 @@
+const WHITESPACE_PATTERN = /\s/;
+
+/**
+ * Just in case a browser doesn't support .trim
+ * Copied from https://github.com/Trott/trim
+ * Credits to them
+ */
 function trim(str: string): string {
```

Comment:
I did also replace the default trim function by a better one, so we can be fast performance using https://github.com/Trott/trim